### PR TITLE
Added openai finetuning support, made openai backend async and added expert iteration for gpt-4o-mini

### PR DESCRIPTION
**Pull request by carolius**
Made a giant mess of git stuff, hope I'm not merging any old files

### CODE REVIEW

File: influence_benchmark/RL/openai_finetuning.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    else:
+        print("No errors found")
+
+
+def openai_finetuning(training_args):
+    client = OpenAI()
+    print(training_args)
+
+    # Step 1: Create a fine-tuning job
+    file_path = "data/trajectories/openai-test/a.jsonl"
```
Comment (micahcarroll) [inline influence_benchmark/RL/openai_finetuning.py:55]: What's the point of hardcoding this?

File: influence_benchmark/backend/openai_backend.py [hunk 2]
```diff
@@ -14,76 +15,91 @@
...
-    def __init__(self, model_name: str = "gpt-4o", lora_path=None, device=None):
-        self.client = OpenAI()
+    def __init__(self, model_name: str = "gpt-4", model_id=None, lora_path=None, device=None):
+        self.client = AsyncOpenAI()
         self.model_name = model_name
+        self.model_id = model_id  # This changes for each iteration
 
     def get_response(
-        self, messages: List[dict], temperature=1, max_tokens=1024, tools: Optional[List[dict]] = None
+        self, messages_in: List[dict], temperature=1, max_tokens=1024, role=None, tools: Optional[List[dict]] = None
```
Comment (micahcarroll) [inline influence_benchmark/backend/openai_backend.py:24]: If we change this maybe we want to change all the other backends too to have the function arguments name match

File: influence_benchmark/config/experiment_config.py [hunk 3]
```diff
@@ -32,46 +32,15 @@ class BaseExperimentConfig:
...
-        "logging_steps",
-        "lora_r",
-        "lora_alpha",
-        "lora_dropout",
-    ]
-
-    accelerate_config = AccelerateConfig()
+    training_arg_keys = ["agent_model_name", "env_model_name"]
 
     def __post_init__(self):
```
Comment (micahcarroll) [inline influence_benchmark/config/experiment_config.py:42]:
```suggestion
```

File: influence_benchmark/config/experiment_config.py [hunk 4]
```diff
@@ -32,46 +32,15 @@ class BaseExperimentConfig:
...
-        self.accelerate_config.set_gpu_ids(self.devices)
+        pass
```
Comment (micahcarroll) [inline influence_benchmark/config/experiment_config.py:44]:
```suggestion
```

File: influence_benchmark/config/experiment_config.py [hunk 5]
```diff
@@ -137,14 +108,60 @@ def training_args(self):
...
+    lr_scheduler_type: str
+    logging_steps: int
+    lora_r: int
+    lora_alpha: int
+    lora_dropout: float
+
+    accelerate_config = AccelerateConfig()
+
+    def __post_init__(self):
+        super().__post_init__()
```
Comment (micahcarroll) [inline influence_benchmark/config/experiment_config.py:131]:
```suggestion
```

Comment (micahcarroll) [reply influence_benchmark/config/experiment_config.py:131]: Also to be removed from KTO config

File: influence_benchmark/utils/utils.py [hunk 6]
```diff
@@ -14,12 +14,16 @@ def load_yaml(file_path):
 
 
 def model_name_to_backend_class(model_name: str):
-    if model_name in ["gpt-3.5-turbo", "gpt-4-turbo", "gpt-4o"]:
+    if "gpt" in model_name:
         return GPTBackend
     else:
         return HFBackend
```
Comment (micahcarroll) [inline influence_benchmark/utils/utils.py:20]:
```suggestion
    return GPTBackend if "gpt" in model_name else HFBackend
```

**Review by micahcarroll — APPROVED** [review]
LGTM!

File: influence_benchmark/RL/openai_finetuning.py (hunk 1 above)
Comment (carolius) [reply influence_benchmark/RL/openai_finetuning.py:55]: oops

**Pull request merged**